src/environment.rs

```diff
@@ -57,25 +57,15 @@ fn create_prefix_location_file(prefix_file: &Path) -> miette::Result<()> {
 ///     1. It verifies that the prefix location is unchanged.
 ///     2. It verifies that the project supports the current platform.
 ///     3. It verifies that the system requirements are met.
-pub fn sanity_check_project(project: &Project) -> miette::Result<()> {
+pub fn sanity_check_project(project: &Project, no_install: &mut bool) -> miette::Result<()> {
```

Comment:
I don't think we do the mutable parameter stuff allot. So I think its better to do this differently. Maybe @tdejager has an opinion.

Reply:
Yeah, I don't like it as well. I changed it in 91b87fb91b8eb92f4989856313f8c49adceec8a2

It is better now but I'm still not sure.

Reply:
I dont think it makes sense to keep these checks in the same function, maybe move the platform check into some kind of `can_install` function or something like that. As it doesn't seem logical to get a bool back from the sanity check that then doesn't tell you if it was sane but only if you can install or not.

Reply:
Makes sense. Applied in 95e84356942cf8aed845d9fa57e7ea620cdd260b

The platform check is a very simple logic so I didn't create a function for it. Let me know what you prefer / how it looks!